### PR TITLE
Certificate Request functionality added

### LINKED ISSUE #652: Add option to download certificate

### PR DESCRIPTION
**Pull request by iamspathan**
Changes: OverviewFragment added receiver and fileProvider

### CODE REVIEW

File: app/src/main/res/xml/paths.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (aggarwalpulkit596) [inline app/src/main/res/xml/paths.xml:1]: is this needed ?

File: app/src/main/java/com/codingblocks/cbonlineapp/mycourse/overview/OverviewFragment.kt [hunk 2]
```diff
@@ -201,4 +226,9 @@ class OverviewFragment : BaseCBFragment(), AnkoLogger {
             invalidate()
         }
     }
+
+    override fun onDestroy() {
+        activity?.unregisterReceiver(receiver)
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/mycourse/overview/OverviewFragment.kt]: requireActivity()

**Review by aggarwalpulkit596 — CHANGES_REQUESTED** [review]

File: app/src/main/res/xml/paths.xml (hunk 1 above)
Comment (iamspathan) [reply app/src/main/res/xml/paths.xml:1]: Yes, I'm using it to specify the path for the file provider.

File: app/src/main/java/com/codingblocks/cbonlineapp/mycourse/overview/OverviewFragment.kt (hunk 2 above)
Comment (iamspathan) [reply app/src/main/java/com/codingblocks/cbonlineapp/mycourse/overview/OverviewFragment.kt]: Commit made for the change

File: app/src/main/java/com/codingblocks/cbonlineapp/util/Certificate.kt [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+import android.app.DownloadManager
+import android.content.*
+import android.net.Uri
+import android.os.Environment
+import android.widget.Toast
+
+object Certificate  {
+
+    fun downloadCertificateAndShow(context: Context, certificateUrl: String, fileName: String) {
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/util/Certificate.kt]: I guess we don't an object for this instead make this a function inside viewmodel

Comment (iamspathan) [reply app/src/main/java/com/codingblocks/cbonlineapp/util/Certificate.kt]: Sure

**Review by aggarwalpulkit596 — APPROVED** [review]

**Comment by hardik-dua** [comment]
is this issue still open?

**Comment by aggarwalpulkit596** [comment]
No this is completed

**Comment by hardik-dua** [comment]
ok cool

**Pull request merged**